### PR TITLE
MJ final review - edits to chapters 2, 3, 4

### PR DESCRIPTION
**Pull request by mariaruth**
a number of small edits, 2 more substantial updates: 
1) removed content on file naming, added to wiki, added wiki link.
2) added references to example .do file throughout text.

### CODE REVIEW

File: chapters/planning-data-work.tex [hunk 1]
```diff
@@ -483,26 +456,33 @@ \subsection{Documenting and organizing code}
 They all come from the principle that code is an output by itself,
 not just a means to an end,
 and should be written thinking of how easy it will be for someone to read it later.
+At the end of this section, we include a template for a Master Script in Stata,
```
Comment (bbdaniels) [inline chapters/planning-data-work.tex]:
```suggestion
At the end of this section, we include a template for a master script in Stata, 
```

File: chapters/planning-data-work.tex [hunk 2]
```diff
@@ -483,26 +456,33 @@ \subsection{Documenting and organizing code}
 They all come from the principle that code is an output by itself,
 not just a means to an end,
 and should be written thinking of how easy it will be for someone to read it later.
+At the end of this section, we include a template for a Master Script in Stata, 
+to provide a concrete example of the required elements and structure.
+Throughout this section, we refer to lines of the example .do file
```
Comment (bbdaniels) [inline chapters/planning-data-work.tex]:
```suggestion
Throughout this section, we refer to lines of the example do file 
```

File: chapters/planning-data-work.tex [hunk 3]
```diff
@@ -483,26 +456,33 @@ \subsection{Documenting and organizing code}
...
 
 Code documentation is one of the main factors that contribute to readability.
 Start by adding a code header to every file.
 A code header is a long \textbf{comment}\sidenote{
   \textbf{Comments:} Code components that have no function,
   but describe in plain language what the code is supposed to do.
 }
-that details the functionality of the entire script.
+that details the functionality of the entire script;
+refer to lines 5-10 in the example .do file.
```
Comment (bbdaniels) [inline chapters/planning-data-work.tex]:
```suggestion
refer to lines 5-10 in the example do file. 
```

File: chapters/planning-data-work.tex [hunk 4]
```diff
@@ -483,26 +456,33 @@ \subsection{Documenting and organizing code}
...
+that details the functionality of the entire script;
+refer to lines 5-10 in the example .do file. 
 This should include simple things such as
 the purpose of the script and the name of the person who wrote it.
 If you are using a version control software,
 the last time a modification was made and the person who made it will be recorded by that software.
 Otherwise, you should include it in the header.
-Finally, use the header to track the inputs and outputs of the script.
+You should always track the inputs and outputs of the script, as well as the uniquely identifying variable;
+refer to lines 49-51 in the example .do file.
```
Comment (bbdaniels) [inline chapters/planning-data-work.tex]:
```suggestion
refer to lines 49-51 in the example do file. 
```

File: chapters/planning-data-work.tex [hunk 5]
```diff
@@ -483,26 +456,33 @@ \subsection{Documenting and organizing code}
...
+You should always track the inputs and outputs of the script, as well as the uniquely identifying variable;
+refer to lines 49-51 in the example .do file. 
 When you are trying to track down which code creates which data set, this will be very helpful.
 While there are other ways to document decisions related to creating code,
 the information that is relevant to understand the code should always be written in the code file.
 
 In the script, alongside the code, are two types of comments that should be included.
-The first type of comment describes what is being done.
+The first type of comment describes what is being done;
+refer to line 35 in the example .do file.
```
Comment (bbdaniels) [inline chapters/planning-data-work.tex]:
```suggestion
refer to line 35 in the example do file.
```

File: chapters/planning-data-work.tex [hunk 6]
```diff
@@ -542,6 +524,7 @@ \subsection{Documenting and organizing code}
 This is an arbitrary limit, just like the standard restriction of each line to 80 characters:
 it seems to be ``enough but not too much'' for most purposes.
 
+\subsection{Working with a Master script}
```
Comment (bbdaniels) [inline chapters/planning-data-work.tex]:
```suggestion
\subsection{Working with a master script}
```

File: chapters/planning-data-work.tex [hunk 7]
```diff
@@ -569,24 +550,28 @@ \subsection{Documenting and organizing code}
 and when it does, it may take time for you to understand what's causing an error.
 The same applies to changes in data sets and results.
 
-To link code, data and outputs, the master script reflects the structure of the \texttt{DataWork} folder in code
-through globals (in Stata) or string scalars (in R).
+To link code, data and outputs, 
+the master script reflects the structure of the \texttt{DataWork} folder in code
+through globals (in Stata) or string scalars (in R); 
+refer to lines 35-40 of the example .do file.
```
Comment (bbdaniels) [inline chapters/planning-data-work.tex]:
```suggestion
refer to lines 35-40 of the example do file. 
```

File: chapters/planning-data-work.tex [hunk 8]
```diff
@@ -569,24 +550,28 @@ \subsection{Documenting and organizing code}
...
 its structure is the same in each team member's computer.
 The only difference between machines should be
-the path to the project root folder, i.e. the highest-level shared folder, which in the context of \texttt{iefolder} is the \texttt{DataWork} folder.
+the path to the project root folder, i.e. the highest-level shared folder, 
+which in the context of \texttt{iefolder} is the \texttt{DataWork} folder.
 This is reflected in the master script in such a way that
 the only change necessary to run the entire code from a new computer
-is to change the path to the project folder to reflect the filesystem and username.
+is to change the path to the project folder to reflect the filesystem and username; 
+refer to lines 27-32 of the example .do file.
```
Comment (bbdaniels) [inline chapters/planning-data-work.tex]:
```suggestion
refer to lines 27-32 of the example do file.
```

File: chapters/planning-data-work.tex [hunk 9]
```diff
@@ -388,11 +388,9 @@ \subsection{Organizing files and folder structures}
 
 The \texttt{DataWork} folder may be created either inside
 an existing project-based folder structure, or it may be created separately.
-It's usually created by the leading RA in agreement with the PI.
-Increasingly, our recommendation is to create the \texttt{DataWork} folder
-separately from the project management materials,
-reserving the ``project folder'' for contracts, Terms of Reference, briefs and other administrative or management work.
-  \index{project folder}
+It is preferable to create the \texttt{DataWork} folder
+in a code folder maintained separately from the project management materials
```
Comment (kbjarkefur) [inline chapters/planning-data-work.tex]:
```suggestion
separately from the project management materials 
```
I am not sure what "in a code folder" means. It should just be separate, right? And in the lines below we explain what we mean by this.

File: chapters/planning-data-work.tex (hunk 2 above)
Comment (kbjarkefur) [reply chapters/planning-data-work.tex]: Stata write them as `do-files` https://www.stata.com/manuals13/u16.pdf

File: chapters/planning-data-work.tex (hunk 3 above)
Comment (kbjarkefur) [reply chapters/planning-data-work.tex]: Stata write them as `do-files` https://www.stata.com/manuals13/u16.pdf

Comment (kbjarkefur) [reply chapters/planning-data-work.tex]: Agree on how to write do-file

File: chapters/planning-data-work.tex (hunk 4 above)
Comment (kbjarkefur) [reply chapters/planning-data-work.tex]: Agree on how to write do-file

File: chapters/planning-data-work.tex (hunk 5 above)
Comment (kbjarkefur) [reply chapters/planning-data-work.tex]: Agree on how to write do-file

Comment (bbdaniels) [reply chapters/planning-data-work.tex]: Official usage is `do-files` so we should use that throughout.

File: chapters/planning-data-work.tex [hunk 10]
```diff
@@ -517,21 +497,23 @@ \subsection{Documenting and organizing code}
 Even you will probably not remember the exact choices that were made in a couple of weeks.
 Therefore, you must document your precise processes in your code.
 
-Code organization means keeping each piece of code in an easily findable location.
+Code organization means keeping each piece of code in an easy-to-find location.
```
Comment (kbjarkefur) [inline chapters/planning-data-work.tex]:
```suggestion
Code organization means keeping each piece of code in an easy-to-find location and naming them in a meaningful way.
```
I think a reference back to file naming should be included in the definition of code organization

File: chapters/planning-data-work.tex [hunk 11]
```diff
@@ -517,21 +497,23 @@ \subsection{Documenting and organizing code}
...
-you can go straight to \texttt{PART 2: Create new variables},
+you can go straight to \texttt{PART 2: Prepare folder paths and define programs},
 instead of reading line by line through the entire code.
 RStudio, for example, makes it very easy to create sections,
 and it compiles them into an interactive script index for you.
 In Stata, you can use comments to create section headers,
-though they're just there to make the reading easier and don't have functionality.
-You should also add an index in the code header by copying and pasting section titles.
+though they're just there to make the reading easier and don't have functionality;
+refer to line 24 of the example .do file.
```
Comment (kbjarkefur) [inline chapters/planning-data-work.tex]: Agree on how to write do-file

File: chapters/planning-data-work.tex [hunk 12]
```diff
@@ -517,21 +497,23 @@ \subsection{Documenting and organizing code}
...
+You should also add an index in the code header by copying and pasting section titles;
+refer to lines 8-10 in the example .do file.
```
Comment (kbjarkefur) [inline chapters/planning-data-work.tex]: Agree on how to write do-file

**Comment by kbjarkefur** [comment]
Fine to merge after addressing all comments!

File: chapters/planning-data-work.tex (hunk 12 above)
Comment (mariaruth) [reply chapters/planning-data-work.tex]: ok updated all to do-file in 7a73864f826d0a9059d2f3240118a1023dfd78c8

**Comment by mariaruth** [comment]
@bbdaniels and @kbjarkefur - changes made.

**Comment by mariaruth** [comment]
Apologies - i forgot to change from my ch2 branch when making edits to 3 and 4, so adding all into this PR and will make a new branch for chapters 5-7.

File: chapters/sampling-randomization-power.tex [hunk 13]
```diff
@@ -50,38 +50,33 @@ \section{Random processes in Stata}
...
+Any process that includes a random component is a random process,
+including sampling, randomization, power calculation simulations, and algorithms like bootstrapping.
 Randomization is challenging and its mechanics are unintuitive for the human brain.
 ``True'' randomization is also nearly impossible to achieve for computers,
 which are inherently deterministic.\sidenote{
   \url{https://www.random.org/randomness}}
+
+\subsection{Implementing random processes reproducibly in Stata}
+
+Reproducibility in statistical programming means that the outputs of random processes
```
Comment (luizaandrade) [inline chapters/sampling-randomization-power.tex]:
```suggestion
Reproducibility in statistical programming means that the outputs of the programs
```

File: chapters/sampling-randomization-power.tex [hunk 14]
```diff
@@ -50,38 +50,33 @@ \section{Random processes in Stata}
...
+can be re-obtained at a future time.\cite{orozco2018make}
```
Comment (luizaandrade) [inline chapters/sampling-randomization-power.tex]:
```suggestion
to be re-obtained at a future time.\cite{orozco2018make}
```

The way it was phrased before made it seem to me that reproducing just the random processes is enough to be considered reproducible. But we can keep it as was if I was the only one who thought so, or if this is not clearer

Comment (mariaruth) [reply chapters/sampling-randomization-power.tex]: I accepted the first suggestion but I don't understand the second. With the second suggestion, the sentence would read "Reproducibility in statistical programming means that the outputs of random processes **to** be re-obtained at a future time". that's not grammatically correct. i think **can** still makes sense here but let me know if i'm missing something.

**Pull request merged**